src/en/zeroscans/src/eu/kanade/tachiyomi/extension/en/zeroscans/ZeroScans.kt

```diff
@@ -0,0 +1,326 @@
+                zsChapters.addAll(newZSChapterPage.chapters)
+                hasMoreResult = newZSChapterPage.hasNextPage
+            }
+
+            zsChapters.map { it.toSChapter(manga) }.let {
+                return Observable.just(it)
+            }
+        } catch (e: Exception) {
+            Log.e("Zero Scans", "Error parsing chapter list", e)
+            throw(e)
+        }
+    }
+
+    private fun zsChapterListRequest(page: Int, manga: SManga): Request {
+        val mangaId = "$baseUrl${manga.url}".toHttpUrl().queryParameter("id")
+        return GET("$baseUrl/$API_PATH/comic/$mangaId/chapters?sort=desc&page=$page")
+    }
+
+    private fun zsChapterListParse(response: Response): ZeroScansChapterPage {
+        return json.decodeFromString<ZeroScansResponseDto<ZeroScansChaptersResponseDto>>(response.body!!.string())
+            .data.let {
+                ZeroScansChapterPage(
+                    it.data,
+                    it.currentPage < it.lastPage
+                )
+            }
+    }
+
+    class ZeroScansChapterPage(
+        val chapters: List<ZeroScansChapterDto>,
+        val hasNextPage: Boolean
+    )
+
+    private fun ZeroScansChapterDto.toSChapter(manga: SManga): SChapter {
+        val comicSlug = "$baseUrl${manga.url}".toHttpUrl().pathSegments[1]
+        val zsChapter = this
+        return SChapter.create().apply {
+            name = "Chapter ${zsChapter.name}"
+            scanlator = zsChapter.group
+            chapter_number = zsChapter.name.toFloat()
+            date_upload = zsHelper.parseChapterUploadDate(zsChapter.createdAt)
+            url = "/comics/$comicSlug/${zsChapter.id}"
+        }
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val chapterUrlPaths = "$baseUrl${chapter.url}".toHttpUrl().pathSegments
+        val mangaSlug = chapterUrlPaths[1]
+        val chapterId = chapterUrlPaths[2]
+        return GET("$baseUrl/$API_PATH/comic/$mangaSlug/chapters/$chapterId")
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        return json.decodeFromString<ZeroScansResponseDto<ZeroScansPageResponseDto>>(response.body!!.string())
+            .data.chapter.getPages()
+    }
+
+    // Fetch Comics, Genres, Statuses and Rankings on creating source
+    init {
+        Single.fromCallable {
+            runCatching { updateComicsData() }
+        }.subscribeOn(Schedulers.io())
+            .observeOn(Schedulers.io())
+            .subscribe({}, {})
+    }
+
+    // Filters
+    override fun getFilterList(): FilterList {
+        return mutableListOf<Filter<*>>().apply {
+            if (genreList.isNotEmpty()) add(GenreFilter(genreList))
+            if (statusList.isNotEmpty()) add(StatusFilter(statusList))
+            add(SortFilter(sortList))
+            add(RankingsHeader())
+            add(RankingsHeader2())
+            add(RankingsFilter(rankingList))
+        }.let { FilterList(it) }
+    }
```

Comment:
It's better to focus on readability of the code.

```suggestion
    override fun getFilterList(): FilterList {
        val filters = mutableListOf()
        
        if (genreList.isNotEmpty()) {
            filters.add(GenreFilter(genreList))
        }
        
        if (statusList.isNotEmpty()) {
            filters.add(StatusFilter(statusList))
        }
        
        filters += listOf(
            SortFilter(sortList),
            RankingsHeader(),
            RankingsHeader2(),
            RankingsFilter(rankingList)
        )
        
        return FilterList(filters)
    }
```